libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -774,9 +774,9 @@ MAV_MISSION_RESULT AP_Mission::mavlink_to_mission_cmd(const mavlink_mission_item
 #ifdef MAV_FRAME_LOCAL
         case MAV_FRAME_LOCAL:                         // local (relative to home position)
             if (copy_location) {
-                cmd.content.location.lat = 1.0e7f*ToDeg(packet.x/
+                cmd.content.location.lat = ToDeg(packet.x/
```

Comment:
I'm not sure this is a correct/valid manipulation? I haven't used the local coordinates before but as I read the documentation they have always come in as meters and the manipulation that was there was mostly correct. All you would need to do is instead of using packet.x you would need (packet.x / 1e4) and leave the multiplication of the final value. As it stands I think you are getting some weird roundoff/conversion problems in there.

I'd welcome someone else checking me on this.